File: Embarcadero/XE5/CPP/SearchContext/dataData.cpp
//---------------------------------------------------------------------------


#pragma hdrstop

#include "dataData.h"
#include "FMX.Dialogs.hpp"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma classgroup "System.Classes.TPersistent"
#pragma resource "*.dfm"
TdtmdlData *dtmdlData;
//---------------------------------------------------------------------------
String FileName()
{
   return(ExtractFilePath(GetModuleName(0))+"iconData.cds");
}
//---------------------------------------------------------------------------
__fastcall TdtmdlData::TdtmdlData(TComponent* Owner)
	: TDataModule(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TdtmdlData::DataModuleCreate(TObject *Sender)
{
	this->cdsIconData->FileName = FileName();

	if (FileExists(this->cdsIconData->FileName)) {
		try {
			cdsIconData->Active = True;
		} catch (...) {
			cdsIconData->Close();
			cdsIconData->CreateDataSet();
		}
	}
	else
	{
		cdsIconData->CreateDataSet();
	}

}

//---------------------------------------------------------------------------
void __fastcall TdtmdlData::cdsIconDataAfterPost(TDataSet *DataSet)
{
	cdsIconData->SaveToFile();
}
//---------------------------------------------------------------------------

File: Embarcadero/XE5/ObjectPascal/SearchContext/dataData.pas
unit dataData;

interface

uses
  {$IFDEF Windows}
   MidasLib,
  {$ENDIF}
  System.SysUtils, System.Classes, Data.DB, Datasnap.DBClient;

type
  TdtmdlData = class(TDataModule)
    cdsIconData: TClientDataSet;
    dsIconData: TDataSource;
    cdsIconDataID: TIntegerField;
    cdsIconDataCategory: TStringField;
    cdsIconDataDescription: TStringField;
    cdsIconDataSearchTerms: TMemoField;
    cdsIconDataIcon: TBlobField;
    procedure DataModuleCreate(Sender: TObject);
    procedure cdsIconDataAfterPost(DataSet: TDataSet);
  private
    { Private declarations }
    function FileName : string;
  public
    { Public declarations }
  end;

var
  dtmdlData: TdtmdlData;

implementation

{%CLASSGROUP 'System.Classes.TPersistent'}

{$R *.dfm}

procedure TdtmdlData.cdsIconDataAfterPost(DataSet: TDataSet);
begin
  cdsIconData.SaveToFile();
end;

procedure TdtmdlData.DataModuleCreate(Sender: TObject);
begin
  cdsIconData.FileName := FileName;

  if not FileExists(FileName) then
    cdsIconData.CreateDataSet
  else
    cdsIconData.Active := True;
end;

function TdtmdlData.FileName: string;
begin
 Result := ExtractFilePath(ParamStr(0))+'iconData.cds';
end;

end.
